packages/react/src/Hero/Hero.tsx

```diff
@@ -1,69 +1,94 @@
-import React, {forwardRef} from 'react'
+import React, {forwardRef, PropsWithChildren, useMemo} from 'react'
 import clsx from 'clsx'
 import styles from './Hero.module.css'
-import {Button} from '../Button'
-import {Heading} from '../Heading'
+import {Button, ButtonBaseProps} from '../Button'
+import {Heading, HeadingProps} from '../Heading'
 import {Text} from '../Text'
 
 import type {BaseProps} from '../component-helpers'
 import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/hero/base.css'
 
-type Action = {
-  text: string
+export type HeroProps = BaseProps<HTMLDivElement> & {
+  align?: 'start' | 'center'
+  children: React.ReactNode | React.ReactNode[] // Maybe use this from `FAQ`? type FAQRootProps = PropsWithChildren<BaseProps<HTMLElement>>
+}
+
+const Root = forwardRef<HTMLDivElement, HeroProps>(({className, align = 'start', children, ...rest}, ref) => {
+  const HeroActions = useMemo(
+    () =>
+      React.Children.toArray(children).filter(
+        child => React.isValidElement(child) && (child.type === HeroPrimaryAction || child.type === HeroSecondaryAction)
+      ),
+    [children]
+  )
+
+  return (
+    <section
+      className={clsx(styles.Hero, styles[`Hero--align-${align}`], className)}
+      ref={ref}
+      aria-labelledby="hero-section-brand-heading"
+      {...rest}
+    >
+      {React.Children.map(children, child => {
+        if (React.isValidElement(child)) {
+          if (child.type !== HeroPrimaryAction && child.type !== HeroSecondaryAction) {
+            return child
+          }
+        }
+      })}
+      <div className={styles['Hero-actions']}>{HeroActions}</div>
+    </section>
+  )
+})
+
+type HeroHeadingProps = {
+  children: React.ReactNode | React.ReactNode[]
```

Comment:
You should be able to use PropsWithChildren to avoid needing to declare children each time.